### ISSUE TITLE
Enable public route that doesn't require authentication

### ISSUE THREAD

**Issue by jasontxf** [Shopify/shopify-app-template-php#411]
## Overview

Allow public routes to pass without having to go through Shopify's authentication/requirements

## Type

- [ ] Changes to existing features

## Motivation

> What inspired this feature request? What problems were you facing?

I'm trying to recreate [Shopify Example App](https://shopify.dev/apps/getting-started/build-app-example/backend#step-3-connect-the-database-and-api-endpoints-to-the-server) in Laravel from scratch since there seems to be only node version available.

I came across the feature where scanning the QR code as a public URL will redirect to our server to update the scan counter.

However, at the moment, it seems as if all routes created will go through Shopify's authentication process and requirements.
So even if I create a route without including the middleware `shopify.auth` it will still fail

```
Route::get('/qrcodes/{code}/scans', [ScanController::class, 'show']);
```

I will see the following error screen

![Xnapper-2023-01-04-16 13 12](https://user-images.githubusercontent.com/1124731/210511622-8441004f-0f58-49d3-8e58-c5b3a56c83fa.png)

Would like to know what is the recommended way to create a public URL?

Thanks for your assistance in advance.

**Comment by jasontxf** [comment]
@rhysnhall Thanks for replying.

So in another word, if I want to visit a public URL, i need to access http://127.0.0.1:54167/qrcodes/{code}/scans instead of the ngrok host provided by vite?

I'm pretty confused why is there a need for a separate server rather than running everything on laravel? 

When deploy for production usage, do we need to setup another tunnel separately from laravel for it to work too?

**Comment by rhysnhall** [comment]
@jasontxf Yep, access whatever url/port your dev server launched on. This is only for dev, though. It is unnecessary in a production environment, and if you configure Nginx correctly, everything will work, including the public URL.

The need for a separate server is for dev only. One is a dev server for React, and the other is for Laravel. As for why they did things this way, I can't say, as I am not associated with Shopify. I saw your issue while trying to add a public route myself and came back to offer the advice once I worked out what was going on.
